tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/APIView/APIViewRevisionTool.cs

```diff
@@ -0,0 +1,216 @@
+
+        return await GetRevisionComments(apiViewUrl, environment);
+    }
+
+    private async Task<APIViewResponse> GetRevisionContent(ParseResult parseResult, CancellationToken ct)
+    {
+        string? apiViewUrl = parseResult.GetValue(apiViewUrlOption);
+        string? environment = parseResult.GetValue(environmentOption);
+        string? outputFile = parseResult.GetValue(outputFileOption);
+        string? contentType = parseResult.GetValue(contentReturnTypeOption);
+
+        if (!validContentTypes.Contains(contentType, StringComparer.OrdinalIgnoreCase))
+        {
+            return new APIViewResponse { ResponseError = $"Invalid content type '{contentType}'. Must be one of: {string.Join(", ", validContentTypes)}." };
+        }
+
+        if (string.IsNullOrEmpty(apiViewUrl))
+        {
+            return new APIViewResponse { ResponseError = "--url must be provided" };
+        }
+
+        (string? revisionId, string? reviewId) = ExtractIdsFromUrl(apiViewUrl);
+        try
+        {
+            string? result = await _apiViewService.GetRevisionContent(revisionId, reviewId, contentType, environment);
+            if (result == null)
+            {
+                return new APIViewResponse { ResponseError = $"Revision content not found" };
+            }
+
+            if (!string.IsNullOrEmpty(outputFile))
+            {
+                await File.WriteAllTextAsync(outputFile, result, ct);
+
+                return new APIViewResponse
+                {
+                    Success = true,
+                    Message = $"Revision content saved to file: {outputFile} ({result.Length:N0} characters)",
+                };
+            }
+
+            return new APIViewResponse
+            {
+                Success = true,
+                Message = $"Revision content retrieved successfully ({result.Length:N0} characters)",
+                Content = result
+            };
+        }
+        catch (ArgumentException ex)
+        {
+            return new APIViewResponse { ResponseError = ex.Message };
+        }
+        catch (Exception ex)
+        {
+            return new APIViewResponse { ResponseError = $"Failed to get revision content: {ex.Message}" };
+        }
+    }
+
+    private (string? revisionId, string? reviewId) ExtractIdsFromUrl(string url)
+    {
+        if (string.IsNullOrWhiteSpace(url))
+        {
+            throw new ArgumentException("Input cannot be null or empty", nameof(url));
+        }
+
+        if (!Uri.TryCreate(url, UriKind.Absolute, out Uri? uri) || (uri.Scheme != "http" && uri.Scheme != "https"))
+        {
+            throw new ArgumentException("Input needs to be a valid APIView URL (e.g., https://apiview.dev/review/{reviewId}?activeApiRevisionId={revisionId})", nameof(url));
+        }
+
+        string? reviewId = null;
+
+        try
+        {
+            // Extract revision ID from query string
+            var query = System.Web.HttpUtility.ParseQueryString(uri.Query);
```

Comment:
Any reason not to use a REGEX here to simplify this?

Reply:
not really it was more to keep the check on the http but that is done in top so I changed this